### ISSUE TITLE
batch/v2alpha1 API does not respect --runtime-config=api/all=true

### ISSUE THREAD

**Issue by jlowdermilk** [kubernetes/kubernetes#32511]
batch/v2alpha1 api does not get registered when passing `--runtime-config=api/all=true` to the apiserver. Instead it has to be manually enabled using `--runtime-config=batch/v2alpha1=true`. If you don't do this, there is still a batch/v2alpha1 api listed, but it has no resources. wat.

```
kubectl api-versions
apps/v1alpha1
authentication.k8s.io/v1beta1
authorization.k8s.io/v1beta1
autoscaling/v1
batch/v1
batch/v2alpha1
certificates.k8s.io/v1alpha1
extensions/v1beta1
policy/v1alpha1
rbac.authorization.k8s.io/v1alpha1
storage.k8s.io/v1beta1
v1

kubectl get scheduledjob
Unable to list "scheduledjobs": the server could not find the requested resource
```

This is both surprising and worrying. As far as I can tell the only place that has all apis enabled is this one line in [`hack/make-rules/test.sh`](https://github.com/kubernetes/kubernetes/blob/master/hack/make-rules/test.sh#L69), but that doesn't exist for 1.3. Do we have documentation anywhere for how to enable all alpha apis?

cc @krousey @lavalamp

**Comment by jlowdermilk** [comment]
So it appears to be special cased in both [apiserver startup](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/server.go#L151) and the e2e runner: https://github.com/kubernetes/kubernetes/pull/30575. We need to fix this and establish a pattern for how alpha APIs should be turned on before this gets even more convoluted.

**Comment by lavalamp** [comment]
@erictune is this something you can fix? Maybe @krousey or @caesarxuchao can consult if needed.

**Comment by krousey** [comment]
This seems really suspicious to me: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/server.go#L151

I have no idea why it's there and what breaks if we remove it. @erictune @soltysh

**Comment by soltysh** [comment]
This line is there to be able to force using `batch/v2alpha1` storage for ScheduledJobs, without that SJ will try to be stored in `batch/v1`. @deads2k have any idea how we can improve this piece?

**Comment by soltysh** [comment]
I'll try to look into the runtime-config setting along with solving #29998.

**Comment by jlowdermilk** [comment]
Also need to get rid of https://github.com/kubernetes/kubernetes/pull/30575/files#diff-1d71a2214070b0cf0548517780d0a98cR269. That line makes in impossible to specify arbitrary RUNTIME_CONFIG for test jobs, as it will be overwritten.

**Comment by soltysh** [comment]
@jlowdermilk I'm all for it, but atm this is the only valid option to enable batch/v2alpha1 to be available for e2e.

**Comment by soltysh** [comment]
Spawned #32661 for removing explicit `batch/v2alpha1` calls from k8s codebase. The main part of this issue was fixed in #32582. Closing.
